js/services/DataActions.ts

```diff
@@ -16,21 +17,69 @@ const DataActions = {
      * @param snapshot
      */
     execute: (action: any, flow: IFlow, snapshot: any) => {
+
+        const objectData = getObjectData(
+            action.objectDataRequest.objectDataType ?
+            action.objectDataRequest.objectDataType.typeElementId :
+            action.objectDataRequest.typeElementId,
+        );
+
         switch (action.crudOperationType.toUpperCase()) {
         case 'LOAD':
-            const objectData = getObjectData(
-                action.objectDataRequest.objectDataType ?
-                action.objectDataRequest.objectDataType.typeElementId :
-                action.objectDataRequest.typeElementId,
-            );
             const filteredObjectData = ObjectData.filter(objectData, action.objectDataRequest.listFilter, action.objectDataRequest.typeElementId);
             const value = snapshot.getValue(action.valueElementToApplyId);
             setStateValue(action.valueElementToApplyId, value.typeElementId, snapshot, filteredObjectData);
             break;
 
         case 'SAVE':
-            // No implemention for saving as the state will already be updated. If we can't connect to the mothership then we can't
-            // Save the data back to the 3rd party data store
+            const valueReferenceToSave = snapshot.getValue(action.valueElementToApplyId);
+            const typeElementId = valueReferenceToSave.typeElementId;
+            const type = typeElementId ? snapshot.metadata.typeElements.find(typeElement => typeElement.id === typeElementId) : null;
+
+            const valueToSave = getStateValue(
+                action.valueElementToApplyId,
+                typeElementId,
+                valueReferenceToSave.contentType,
+                null,
+            );
+
+            valueToSave.objectData.map((obj) => {
+
+                const existsInCache = objectData.find(
+                    existingObj => existingObj.externalId === obj.externalId,
+                );
+
+                const newObject = [{
+                    typeElementId,
+                    externalId: existsInCache ? existsInCache.externalId : null,
+                    internalId: existsInCache ? existsInCache.internalId : guid(),
+                    developerName: obj.developerName,
+                    order: 0,
+                    isSelected: false,
+                    properties: clone(type.properties).map((property) => {
```

Comment:
Is clone necessary now we have ramda available?

Reply:
Yep, done